### PR TITLE
ui: Dependency Upgrade

### PR DESCRIPTION
**Pull request by johncowen**
This PR is ontop of our ember update to 3.8 and upgrades everything else (apart from ember-exam)

Interesting things of note:

1. We switched all the new lint errors to warnings, we basically put off fixing these until a later date. Running the app now gives you lots of yellow warning messages to annoy, so they will be looked at fairly soon.
2. We removed `dart-sass` which we weren't using but was left in there from an experiment from many moons ago 😮 !
3. We removed a few other things which we don't use but are probably there from when we originally copied our starting skeleton from elsewhere.
4. The `jquery` resolution is no longer needed, the upgrade pushes us into 3.4 (plus we hope to remove jquery entirely soon)
5. We use a slightly custom href-to helper to globally fix wildcard URL encoding for ember wildcard routes (see https://github.com/hashicorp/consul/pull/5206). This needed altering slightly as a result of the upgrade, as a bonus we're aware that ember-href-to now supports sticky query params 🎉 

We've left `ember-exam` without upgrading here potentially @alvin-huang might want to do that, or at least I need to check with him that the upgrade will be good for CI - probably to be done in a later PR.

### CODE REVIEW

File: ui-v2/app/components/templated-anchor.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/*eslint ember/require-return-from-computed: "warn"*/
+// TODO: Remove ^
+
```
Comment (backspace) [inline ui-v2/app/components/templated-anchor.js]: I see that you want to address these later but I think this one would be as simple as an `else { return undefined; }`

**Comment by alvin-huang** [comment]
I can follow up with you on the `ember-exam` upgrade, without testing the command myself and just looking at the latest releases, it looks like the current command in CircleCI should _just work_ still.

**Comment by johncowen** [comment]
@alvin-huang ah thanks for looking, thats cool then if it all looks good to you, I added a commit on the end here, everything continues to pass 🎉 , thankyou!

**Comment by johncowen** [comment]
Closing in favour of https://github.com/hashicorp/consul/pull/6452

**Pull request closed without merging**